src/endpoints/payments.ts

```diff
@@ -1,10 +1,15 @@
 import Endpoint from './base'
-import { PaymentEligibilityPayload } from '../types'
+import { MultipleEligibilityPayload, PaymentEligibilityPayload } from '../types'
 import Eligibility from '../entities/eligibility'
 
 const PAYMENTS_PATH = '/v1/payments'
 
+type EligibilityResult<
+  T extends PaymentEligibilityPayload
+> = T['payment'] extends MultipleEligibilityPayload ? Eligibility[] : Eligibility
+
 export class PaymentsEndpoint extends Endpoint {
+  async eligibility<T extends PaymentEligibilityPayload>(data: T): Promise<EligibilityResult<T>>
   async eligibility(data: PaymentEligibilityPayload): Promise<Eligibility | Eligibility[]> {
```

Comment:
The additional line is not buggy: it's an overload for the actual method, with a generic type to let TypeScript decide between `Eligibility` and `Eligibility[]` based on the `EligibilityResult` conditional type